### ISSUE TITLE
[Rule] Detect the use of .then(function(){}, function(){})

### ISSUE THREAD

**Issue by Gillespie59** [Gillespie59/eslint-plugin-angular#227]
When using promise, we often use the `then` method with two parameters : one for success and one for error. Some guidelines explain that we should avoid the use of this signature, and use the `catch` method. 

```javascript
//bad
$http(...).then(
    function(){},
    function(){},
)

//good
$http(...)
    .then(function(){})
    .catch(function(){})
```

**Comment by jrencz** [comment]
@Gillespie59 in fact the analog of the first case is 

```js
$http(...)
    .catch(function(){})
    .then(function(){})
```

I think this rule is more general: it's not about angular, it's about promises in general.

There are promise plugins already:
- [jden/eslint-plugin-promiseparams](https://github.com/jden/eslint-plugin-promiseparams)
- [jden/xjamundx/eslint-plugin-promise](https://github.com/xjamundx/eslint-plugin-promise) (fork of jden/eslint-plugin-promiseparams)
